### PR TITLE
Compare pyg VS pyg and graphbolt

### PR DESCRIPTION
**Pull request by hutiechuan**
## Description

python '/home/ubuntu/dgl/examples/sampling/pyg/pure_pyg.py' 
device is:  cpu
Epoch: 0, Loss: 2.1602, Accuracy: 0.4013, Val Acc: 0.4994
Epoch: 1, Loss: 1.5205, Accuracy: 0.5548, Val Acc: 0.5373
Epoch: 2, Loss: 1.4128, Accuracy: 0.5814, Val Acc: 0.5565
Epoch: 3, Loss: 1.3589, Accuracy: 0.5941, Val Acc: 0.5718
Epoch: 4, Loss: 1.3307, Accuracy: 0.6025, Val Acc: 0.5734
Epoch: 5, Loss: 1.3063, Accuracy: 0.6090, Val Acc: 0.5736
Epoch: 6, Loss: 1.2928, Accuracy: 0.6120, Val Acc: 0.5742
Epoch: 7, Loss: 1.2833, Accuracy: 0.6161, Val Acc: 0.5869
Epoch: 8, Loss: 1.2721, Accuracy: 0.6184, Val Acc: 0.5864
Epoch: 9, Loss: 1.2595, Accuracy: 0.6212, Val Acc: 0.5899
Test Accuracy: 0.5899
Total Training Time: 81.45107126235962 seconds
Peak Memory Usage: 525.70703125 MiB

python '/home/ubuntu/dgl/examples/sampling/pyg/pure_pyg.py'
device is:  cuda
Epoch: 0, Loss: 2.0674, Accuracy: 0.4229, Val Acc: 0.5168
Epoch: 1, Loss: 1.5057, Accuracy: 0.5598, Val Acc: 0.5413
Epoch: 2, Loss: 1.4050, Accuracy: 0.5844, Val Acc: 0.5549
Epoch: 3, Loss: 1.3579, Accuracy: 0.5939, Val Acc: 0.5678
Epoch: 4, Loss: 1.3345, Accuracy: 0.6023, Val Acc: 0.5792
Epoch: 5, Loss: 1.3141, Accuracy: 0.6057, Val Acc: 0.5727
Epoch: 6, Loss: 1.2889, Accuracy: 0.6129, Val Acc: 0.5777
Epoch: 7, Loss: 1.2736, Accuracy: 0.6161, Val Acc: 0.5824
Epoch: 8, Loss: 1.2701, Accuracy: 0.6172, Val Acc: 0.5836
Epoch: 9, Loss: 1.2555, Accuracy: 0.6219, Val Acc: 0.5844
Test Accuracy: 0.5844
Total Training Time: 26.798791885375977 seconds
Peak Memory Usage: 729.98046875 MiB
 

 python '/home/ubuntu/dgl/examples/sampling/pyg/pure_pyg.py' --dataset ogbn-products
device is:  cpu
Epoch: 0, Loss: 0.6574, Accuracy: 0.8288, Val Acc: 0.6983
Epoch: 1, Loss: 0.4471, Accuracy: 0.8786, Val Acc: 0.6877
Epoch: 2, Loss: 0.4253, Accuracy: 0.8838, Val Acc: 0.6529
Epoch: 3, Loss: 0.4319, Accuracy: 0.8806, Val Acc: 0.7069
Epoch: 4, Loss: 0.4023, Accuracy: 0.8885, Val Acc: 0.6191
Epoch: 5, Loss: 0.4298, Accuracy: 0.8831, Val Acc: 0.6458
Epoch: 6, Loss: 0.4202, Accuracy: 0.8865, Val Acc: 0.6557
Epoch: 7, Loss: 0.4206, Accuracy: 0.8846, Val Acc: 0.7199
Epoch: 8, Loss: 0.3881, Accuracy: 0.8917, Val Acc: 0.7159
Epoch: 9, Loss: 0.3816, Accuracy: 0.8945, Val Acc: 0.7113
Test Accuracy: 0.7113
Total Training Time: 8726.167247533798 seconds
Peak Memory Usage: 10977.609375 MiB

 python '/home/ubuntu/dgl/examples/sampling/pyg/pure_pyg.py' --dataset ogbn-products
device is:  cuda
/opt/conda/envs/dgl-dev-gpu-121/lib/python3.8/site-packages/torch/cuda/memory.py:329: FutureWarning: torch.cuda.reset_max_memory_allocated now calls torch.cuda.reset_peak_memory_stats, which resets /all/ peak memory stats.
  warnings.warn(
Before training - GPU Allocated Memory: 2979198976 bytes
Epoch: 0, Loss: 0.6590, Accuracy: 0.8293, Val Acc: 0.6701
Epoch: 1, Loss: 0.4594, Accuracy: 0.8763, Val Acc: 0.6105
Epoch: 2, Loss: 0.4273, Accuracy: 0.8830, Val Acc: 0.6304
Epoch: 3, Loss: 0.4226, Accuracy: 0.8823, Val Acc: 0.7155
Epoch: 4, Loss: 0.3962, Accuracy: 0.8901, Val Acc: 0.6724
Epoch: 5, Loss: 0.3972, Accuracy: 0.8896, Val Acc: 0.6844
Epoch: 6, Loss: 0.3914, Accuracy: 0.8909, Val Acc: 0.7265
Epoch: 7, Loss: 0.3811, Accuracy: 0.8937, Val Acc: 0.6583
Epoch: 8, Loss: 0.3987, Accuracy: 0.8903, Val Acc: 0.6593
Epoch: 9, Loss: 0.4012, Accuracy: 0.8899, Val Acc: 0.7032
Test Accuracy: 0.7032
Total Training Time: 993.9963409900665 seconds
Peak Memory Usage: 14796.27734375 MiB
After training - Allocated Memory: 2997089280 bytes
Peak GPU Memory Usage: 5402692608 bytes 
GPU: 2.42 GB

python '/home/ubuntu/dgl/examples/sampling/pyg/node_classification.py' --dataset ogbn-products
The dataset is already preprocessed.
devic is:  cuda
Dataloader Creation Time: 0.0076792240142822266 seconds
Epoch 0, Train Loss: 0.0000, Train Accuracy: 0.8313
Valid Accuracy: 0.8617
Epoch 1, Train Loss: 0.0000, Train Accuracy: 0.8789
Valid Accuracy: 0.8753
Epoch 2, Train Loss: 0.0000, Train Accuracy: 0.8867
Valid Accuracy: 0.8800
Epoch 3, Train Loss: 0.0000, Train Accuracy: 0.8878
Valid Accuracy: 0.8782
Epoch 4, Train Loss: 0.0000, Train Accuracy: 0.8926
Valid Accuracy: 0.8746
Epoch 5, Train Loss: 0.0000, Train Accuracy: 0.8923
Valid Accuracy: 0.8782
Epoch 6, Train Loss: 0.0000, Train Accuracy: 0.8936
Valid Accuracy: 0.8776
Epoch 7, Train Loss: 0.0000, Train Accuracy: 0.8941
Valid Accuracy: 0.8776
Epoch 8, Train Loss: 0.0000, Train Accuracy: 0.8951
Valid Accuracy: 0.8832
Epoch 9, Train Loss: 0.0000, Train Accuracy: 0.8948
Valid Accuracy: 0.8831
Test Accuracy: 0.8842
Total Training Time: 466.7219305038452 seconds
Peak Memory Usage: 2417.5390625 MiB
Initial Memory: 0.0 MB
Final Memory: 17.85791015625 MB
Peak Memory during Training: 2329.625 MB

python '/home/ubuntu/dgl/examples/sampling/pyg/node_classification.py'
The dataset is already preprocessed.
devic is:  cuda
Dataloader Creation Time: 0.007941961288452148 seconds
Epoch 0, Train Loss: 0.0000, Train Accuracy: 0.4242
Valid Accuracy: 0.5191
Epoch 1, Train Loss: 0.0000, Train Accuracy: 0.5626
Valid Accuracy: 0.5578
Epoch 2, Train Loss: 0.0000, Train Accuracy: 0.5857
Valid Accuracy: 0.5760
Epoch 3, Train Loss: 0.0000, Train Accuracy: 0.5974
Valid Accuracy: 0.5812
Epoch 4, Train Loss: 0.0000, Train Accuracy: 0.6013
Valid Accuracy: 0.5814
Epoch 5, Train Loss: 0.0000, Train Accuracy: 0.6082
Valid Accuracy: 0.5906
Epoch 6, Train Loss: 0.0000, Train Accuracy: 0.6164
Valid Accuracy: 0.5893
Epoch 7, Train Loss: 0.0000, Train Accuracy: 0.6170
Valid Accuracy: 0.6029
Epoch 8, Train Loss: 0.0000, Train Accuracy: 0.6230
Valid Accuracy: 0.5971
Epoch 9, Train Loss: 0.0000, Train Accuracy: 0.6227
Valid Accuracy: 0.5970
Test Accuracy: 0.5969
otal Training Time: 17.793312788009644 seconds
Peak Memory Usage: 808.671875 MiB
Initial Memory: 0.0 MB
Final Memory: 17.4130859375 MB
Peak Memory during Training: 150.8271484375 MB

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by caojy1998** [comment]
Please paste the result of the code in description.
